Docs/DevelopingHastlayer.md

```diff
@@ -8,28 +8,10 @@ How to work on Hastlayer itself?
 - Software that was written previously, without knowing about Hastlayer should be usable if it can live within the constraints of transformable code. E.g. users should never be forced to use custom attributes or other Hastlayer-specific elements in their code if the same effect can be achieved with runtime configuration (think about how members to be processed are configured: when running Hastlayer, not with attributes).
 - If some code uses unsupported constructs it should be made apparent with exceptions. The hardware implementation silently failing (or working unexpectedly) should be avoided. Exceptions should include contextual information (e.g. the full expression or method of the problematic source, link to the GitHub issue on adding support for it; see the `AddParentEntityName()` extension method) and offer hints on possible solutions.
 
-## Maintaining the flavors of the Hastlayer solution
-
-Generally the *client* branch should be only merged to, but never from, the *dev* branch.
-
-There are separate solution files for the two flavors that only differ in whether they include *Hast.Core*. Should the solution change then make the changes in the dev solution file, copy it over the client one and remove the *Hast.Core* solution folder. If you switch between the two solutions while on the *dev* branch then temporarily configure the Hastlayer shell to use the Client flavor (see `IHastlayerConfiguration`).
-
-To allow the same code in the samples and elsewhere to support both scenarios dynamic library loading is utilized. For this to work *Hast.Core* projects should adhere to the following:
-
-- The built DLL name to start with "Hast."
-- Their projects need to be located in a subfolder of *Hast.Core*.
-- Both the Debug and Release build output directories should be set just to *bin\\*.
-
-If a *Hast.Core* projects needs to have types accessible in the Client flavor then create a corresponding `Abstractions` project. Such projects should follow the same rules listed above as *Hast.Core* projects with the only difference being that they should be located in a subfolder of *Hast.Abstractions*. Exceptions are projects that are directly added as imported extensions in `Hast.Layer`: Those can be just normal projects (like `Hast.Transformer.Abstractions`).
```

Comment:
These are not applicable anymore, but `GetHastLibraries()` still loads dependencies from the available (hard-referenced) Hast DLLs. We can perhaps change that to use static loading, but for that, we'd need to add a reference to a type in each assembly, which seems like a chore and error-prone in the future if we add new projects.

Reply:
I wonder if we can determine the types using some kind of compile time analysis using Roslyn code generation? Then this could be automated away and we'd have one less error source.

Reply:
We can iterate over the loaded assemblies instead of the files too, and filter by name there. Apart from the name, we could use a marker attribute in `SharedAssemblyInfo`, but I don't think that's much better.

Reply:
Based on https://github.com/Lombiq/Hastlayer-SDK/issues/99 I suggest leaving this as it is for now.

Reply:
Yeah, no change request here. I just answered your question where you asked. 😉